crates/sui-node/src/lib.rs

```diff
@@ -999,6 +997,22 @@ impl SuiNode {
     }
 }
 
+/// Notify state-sync that a new list of trusted peers are now available.
+fn send_trusted_peer_change(
+    config: &NodeConfig,
+    sender: &watch::Sender<TrustedPeerChangeEvent>,
+    epoch_state_state: &EpochStartSystemState,
+) {
+    if let Err(err) = sender.send(TrustedPeerChangeEvent {
+        new_peers: epoch_state_state.get_validator_as_p2p_peers(config.protocol_public_key()),
+    }) {
+        warn!(
```

Comment:
When can this fail, and should we log error or panic since it seems pretty bad when TrustedPeerChangeEvent is not updated?

Reply:
It can only fail when the receiver is closed. But I wouldn't say it's panic-level bad because validators don't change often. So even if we fail to send over it should affect the network much in majority of cases (maybe except the initial one)